### PR TITLE
Rhfogh mxcube3

### PR DESCRIPTION
**Pull request by rhfogh**
GPhL-specific changs, and a few bug fixes in common code.

### CODE REVIEW

File: mxcubecore/HardwareObjects/abstract/AbstractCollect.py [hunk 1]
```diff
@@ -24,6 +24,7 @@
 """
 
 import os
+import sys
```
Comment (IvarsKarpics) [inline mxcubecore/HardwareObjects/abstract/AbstractCollect.py:27]: Is there a need for sys import?

File: mxcubecore/HardwareObjects/Gphl/GphlMessages.py [hunk 2]
```diff
@@ -816,12 +816,7 @@ def __init__(self, data_model):
             self._cell = UnitCell(*cell_parameters)
         else:
             self._cell = None
-        detector_setting = (
-            data_model.char_detector_setting or data_model.detector_setting
-        )
-        self._expectedResolution = (
-            detector_setting.resolution if detector_setting else None
-        )
+        self._expectedResolution = data_model.aimed_resolution
```
Comment (IvarsKarpics) [inline mxcubecore/HardwareObjects/Gphl/GphlMessages.py:819]: What about the variable naming. This looks like a combination of snake and camel cases. Could it be just in snake case?

**Review by IvarsKarpics — CHANGES_REQUESTED** [review]

File: mxcubecore/HardwareObjects/abstract/AbstractCollect.py (hunk 1 above)
Comment (rhfogh) [reply mxcubecore/HardwareObjects/abstract/AbstractCollect.py:27]: There is  call to sys somewhere in error handling lower down

File: mxcubecore/HardwareObjects/GphlWorkflow.py [hunk 3]
```diff
@@ -1544,6 +1558,9 @@ def auto_select_solution(self, choose_lattice):
                     lattice_fit = line
         useline = lattice_fit or system_fit or starred
         if useline:
+            logging.getLogger("user_level_log").info(
+                "Selected iudexing solution", useline
```
Comment (beteva) [inline mxcubecore/HardwareObjects/GphlWorkflow.py:1562]: iudexing?

File: mxcubecore/HardwareObjects/GphlWorkflow.py [hunk 4]
```diff
@@ -1559,6 +1576,18 @@ def select_lattice(self, payload, correlation_id):
 
         if data_model.automation_mode:
             solution = self.auto_select_solution(choose_lattice)
+
+            if data_model.automation_mode == "MASSIF1":
```
Comment (beteva) [inline mxcubecore/HardwareObjects/GphlWorkflow.py:1580]: Has MASSIF1 become a common noun? 😃

**Pull request merged**